// @whop/sdk/lib/verify-user-token.mjs
import { createRemoteJWKSet, importJWK, jwtVerify } from 'jose';
const USER_TOKEN_HEADER_NAME = 'x-whop-user-token';
const DEFAULT_JWKS_URL = 'https://api.whop.com/.well-known/jwks.json';
// Module-level cache: one RemoteJWKSet per URL. jose's remote set handles
// HTTP caching (respecting the endpoint's Cache-Control), a cooldown to
// prevent tight-loop refetches on unknown kids, and thread-safe in-flight
// deduplication. Reusing the same instance across calls is what makes the
// verify path effectively free after the first lookup.
const jwksCache = new Map();
function getRemoteJwks(url) {
    let existing = jwksCache.get(url);
    if (existing)
        return existing;
    existing = createRemoteJWKSet(new URL(url), {
        cacheMaxAge: 12 * 60 * 60 * 1000,
        cooldownDuration: 30000,
    });
    jwksCache.set(url, existing);
    return existing;
}
export function getUserToken(tokenOrHeadersOrRequest, options) {
    const headerName = options?.headerName ?? USER_TOKEN_HEADER_NAME;
    if (typeof tokenOrHeadersOrRequest === 'string')
        return tokenOrHeadersOrRequest;
    if (tokenOrHeadersOrRequest instanceof Headers)
        return tokenOrHeadersOrRequest.get(headerName);
    if (tokenOrHeadersOrRequest instanceof Request)
        return tokenOrHeadersOrRequest.headers.get(headerName);
    return null;
}
export function makeUserTokenVerifierFromSdk(client) {
    return async function verifyUserToken(tokenOrHeadersOrRequest, options) {
        if (!client.appID) {
            throw Error('You must set appID in the Whop client constructor if you want to verify user tokens.');
        }
        const baseOptions = { appId: client.appID };
        if (client.userTokenPublicKey)
            baseOptions.publicKey = client.userTokenPublicKey;
        if (client.userTokenJwksUrl)
            baseOptions.jwksUrl = client.userTokenJwksUrl;
        return await internalVerifyUserToken(tokenOrHeadersOrRequest, {
            ...baseOptions,
            ...options,
        });
    };
}
export function verifyUserToken(tokenOrHeadersOrRequest, overrideOptions) {
    return internalVerifyUserToken(tokenOrHeadersOrRequest, {
        ...overrideOptions,
    });
}
async function internalVerifyUserToken(tokenOrHeadersOrRequest, options) {
    try {
        const tokenString = getUserToken(tokenOrHeadersOrRequest, {
            headerName: options?.headerName,
        });
        if (!tokenString) {
            throw new Error('Whop user token not found. If you are the app developer, ensure you are developing in the whop.com iframe and have the dev proxy enabled.');
        }
        const verifyOptions = { issuer: 'urn:whopcom:exp-proxy', algorithms: ['ES256'] };
        let token;
        if (options.publicKey) {
            const key = await importJWK(JSON.parse(options.publicKey), 'ES256').catch(() => {
                throw new Error('Invalid public key provided to verifyUserToken');
            });
            token = await jwtVerify(tokenString, key, verifyOptions).catch(() => {
                throw new Error('Invalid user token provided to verifyUserToken');
            });
        }
        else {
            const jwks = getRemoteJwks(options.jwksUrl ?? DEFAULT_JWKS_URL);
            token = await jwtVerify(tokenString, jwks, verifyOptions).catch(() => {
                throw new Error('Invalid user token provided to verifyUserToken');
            });
        }
        if (!(token.payload.sub && token.payload.aud) || Array.isArray(token.payload.aud)) {
            throw new Error('Invalid user token provided to verifyUserToken');
        }
        if (options.appId && token.payload.aud !== options.appId)
            throw new Error('Invalid app id provided to verifyUserToken');
        return {
            appId: token.payload.aud,
            userId: token.payload.sub,
        };
    }
    catch (e) {
        if (options.dontThrow) {
            return null;
        }
        throw e;
    }
}
