// @whop/sdk/resources/accounts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
export class Accounts extends APIResource {
    /**
     * Lists accounts visible to the credential. User tokens return the user's business
     * accounts; business account API keys return the requesting business account and
     * its connected accounts.
     */
    list(query = {}, options) {
        return this._client.get('/accounts', { query, ...options });
    }
    /**
     * Creates an account. User tokens create business accounts; business account API
     * keys create connected accounts.
     */
    create(body, options) {
        return this._client.post('/accounts', { body, ...options });
    }
    /**
     * Retrieves the business account associated with the current business account API
     * key.
     */
    me(options) {
        return this._client.get('/accounts/me', options);
    }
    /**
     * Retrieves a single account visible to the credential, including its crypto
     * wallet.
     */
    retrieve(accountID, options) {
        return this._client.get(path `/accounts/${accountID}`, options);
    }
    /**
     * Updates an account. User tokens can update business accounts; business account
     * API keys can update connected accounts.
     */
    update(accountID, body, options) {
        return this._client.patch(path `/accounts/${accountID}`, { body, ...options });
    }
}
