// @whop/sdk/resources/ad-campaigns.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Ad campaigns
 */
export class AdCampaigns extends APIResource {
    /**
     * Returns a paginated list of ad campaigns for a company, with optional filtering
     * by status, and creation date.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const adCampaignListResponse of client.adCampaigns.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ad_campaigns', (CursorPage), {
            query,
            ...options,
        });
    }
    /**
     * Retrieves a single ad campaign by its unique identifier.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.retrieve(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/ad_campaigns/${id}`, options);
    }
    /**
     * Updates an ad campaign synchronously.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.update(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/ad_campaigns/${id}`, { body, ...options });
    }
    /**
     * Pauses an ad campaign, optionally until a specific date.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.pause(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    pause(id, options) {
        return this._client.post(path `/ad_campaigns/${id}/pause`, options);
    }
    /**
     * Resumes a paused ad campaign.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.unpause(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    unpause(id, options) {
        return this._client.post(path `/ad_campaigns/${id}/unpause`, options);
    }
}
