// @whop/sdk/resources/ad-campaigns.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdCampaigns = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Ad campaigns
 */
class AdCampaigns extends resource_1.APIResource {
    /**
     * Returns a paginated list of ad campaigns for a company, with optional filtering
     * by status, and creation date.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const adCampaignListResponse of client.adCampaigns.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ad_campaigns', (pagination_1.CursorPage), {
            query,
            ...options,
        });
    }
    /**
     * Retrieves a single ad campaign by its unique identifier.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.retrieve(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/ad_campaigns/${id}`, options);
    }
    /**
     * Updates an ad campaign synchronously.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.update(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/ad_campaigns/${id}`, { body, ...options });
    }
    /**
     * Pauses an ad campaign, optionally until a specific date.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.pause(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    pause(id, options) {
        return this._client.post((0, path_1.path) `/ad_campaigns/${id}/pause`, options);
    }
    /**
     * Resumes a paused ad campaign.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adCampaign = await client.adCampaigns.unpause(
     *   'adcamp_xxxxxxxxxxx',
     * );
     * ```
     */
    unpause(id, options) {
        return this._client.post((0, path_1.path) `/ad_campaigns/${id}/unpause`, options);
    }
}
exports.AdCampaigns = AdCampaigns;
