// @whop/sdk/resources/ad-groups.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroups = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Ad groups
 */
class AdGroups extends resource_1.APIResource {
    /**
     * Returns a paginated list of ad groups scoped by campaign or company, with
     * optional filtering by status and creation date.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const adGroupListResponse of client.adGroups.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ad_groups', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Retrieves a single ad group by its unique identifier.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.retrieve(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/ad_groups/${id}`, options);
    }
    /**
     * Updates an existing ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.update(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/ad_groups/${id}`, { body, ...options });
    }
    /**
     * Soft-deletes an ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.delete(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/ad_groups/${id}`, options);
    }
    /**
     * Pauses an ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.pause(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    pause(id, options) {
        return this._client.post((0, path_1.path) `/ad_groups/${id}/pause`, options);
    }
    /**
     * Resumes a paused ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.unpause(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    unpause(id, options) {
        return this._client.post((0, path_1.path) `/ad_groups/${id}/unpause`, options);
    }
}
exports.AdGroups = AdGroups;
