// @whop/sdk/resources/ad-groups.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Ad groups
 */
export class AdGroups extends APIResource {
    /**
     * Returns a paginated list of ad groups scoped by campaign or company, with
     * optional filtering by status and creation date.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const adGroupListResponse of client.adGroups.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ad_groups', (CursorPage), { query, ...options });
    }
    /**
     * Retrieves a single ad group by its unique identifier.
     *
     * Required permissions:
     *
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.retrieve(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/ad_groups/${id}`, options);
    }
    /**
     * Updates an existing ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.update(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/ad_groups/${id}`, { body, ...options });
    }
    /**
     * Soft-deletes an ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.delete(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/ad_groups/${id}`, options);
    }
    /**
     * Pauses an ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.pause(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    pause(id, options) {
        return this._client.post(path `/ad_groups/${id}/pause`, options);
    }
    /**
     * Resumes a paused ad group.
     *
     * Required permissions:
     *
     * - `ad_campaign:update`
     * - `ad_campaign:basic:read`
     *
     * @example
     * ```ts
     * const adGroup = await client.adGroups.unpause(
     *   'adgrp_xxxxxxxxxxxx',
     * );
     * ```
     */
    unpause(id, options) {
        return this._client.post(path `/ad_groups/${id}/unpause`, options);
    }
}
