// @whop/sdk/resources/ai-chats.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AIChats = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Ai chats
 */
class AIChats extends resource_1.APIResource {
    /**
     * Returns a paginated list of AI chat threads for the current authenticated user.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai_chats', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Create a new AI chat thread and send the first message to the AI agent.
     *
     * Required permissions:
     *
     * - `ai_chat:create`
     */
    create(body, options) {
        return this._client.post('/ai_chats', { body, ...options });
    }
    /**
     * Retrieves the details of an existing AI chat.
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/ai_chats/${id}`, options);
    }
    /**
     * Update an AI chat's title, notification preferences, or associated company
     * context.
     *
     * Required permissions:
     *
     * - `ai_chat:update`
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/ai_chats/${id}`, { body, ...options });
    }
    /**
     * Delete an AI chat thread so it no longer appears in the user's chat list.
     *
     * Required permissions:
     *
     * - `ai_chat:delete`
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/ai_chats/${id}`, options);
    }
}
exports.AIChats = AIChats;
