// @whop/sdk/resources/ai-chats.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Ai chats
 */
export class AIChats extends APIResource {
    /**
     * Returns a paginated list of AI chat threads for the current authenticated user.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai_chats', (CursorPage), { query, ...options });
    }
    /**
     * Create a new AI chat thread and send the first message to the AI agent.
     *
     * Required permissions:
     *
     * - `ai_chat:create`
     */
    create(body, options) {
        return this._client.post('/ai_chats', { body, ...options });
    }
    /**
     * Retrieves the details of an existing AI chat.
     */
    retrieve(id, options) {
        return this._client.get(path `/ai_chats/${id}`, options);
    }
    /**
     * Update an AI chat's title, notification preferences, or associated company
     * context.
     *
     * Required permissions:
     *
     * - `ai_chat:update`
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/ai_chats/${id}`, { body, ...options });
    }
    /**
     * Delete an AI chat thread so it no longer appears in the user's chat list.
     *
     * Required permissions:
     *
     * - `ai_chat:delete`
     */
    delete(id, options) {
        return this._client.delete(path `/ai_chats/${id}`, options);
    }
}
