// @whop/sdk/resources/companies.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Companies
 */
export class Companies extends APIResource {
    /**
     * Retrieves the details of an existing company.
     *
     * Required permissions:
     *
     * - `company:basic:read`
     */
    retrieve(id, options) {
        return this._client.get(path `/companies/${id}`, options);
    }
    /**
     * Returns a paginated list of companies. When parent_company_id is provided, lists
     * connected accounts under that platform. When omitted, lists companies the
     * current user has access to.
     *
     * Required permissions:
     *
     * - `company:basic:read`
     */
    list(query = {}, options) {
        return this._client.getAPIList('/companies', (CursorPage), { query, ...options });
    }
    /**
     * Create a new company. Pass parent_company_id to create a connected account under
     * a platform, or omit it to create a company for the current user.
     *
     * Required permissions:
     *
     * - `company:create`
     * - `company:basic:read`
     */
    create(body, options) {
        return this._client.post('/companies', { body, ...options });
    }
    /**
     * Update a company's title, description, logo, and other settings.
     *
     * Required permissions:
     *
     * - `company:update`
     * - `company:basic:read`
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/companies/${id}`, { body, ...options });
    }
    /**
     * Create an API key for a connected account (child company) owned by a parent
     * company.
     */
    createAPIKey(parentCompanyID, body, options) {
        return this._client.post(path `/companies/${parentCompanyID}/api_keys`, { body, ...options });
    }
}
