// @whop/sdk/resources/companies.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Companies = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Companies
 */
class Companies extends resource_1.APIResource {
    /**
     * Retrieves the details of an existing company.
     *
     * Required permissions:
     *
     * - `company:basic:read`
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/companies/${id}`, options);
    }
    /**
     * Returns a paginated list of companies. When parent_company_id is provided, lists
     * connected accounts under that platform. When omitted, lists companies the
     * current user has access to.
     *
     * Required permissions:
     *
     * - `company:basic:read`
     */
    list(query = {}, options) {
        return this._client.getAPIList('/companies', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Create a new company. Pass parent_company_id to create a connected account under
     * a platform, or omit it to create a company for the current user.
     *
     * Required permissions:
     *
     * - `company:create`
     * - `company:basic:read`
     */
    create(body, options) {
        return this._client.post('/companies', { body, ...options });
    }
    /**
     * Update a company's title, description, logo, and other settings.
     *
     * Required permissions:
     *
     * - `company:update`
     * - `company:basic:read`
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/companies/${id}`, { body, ...options });
    }
    /**
     * Create an API key for a connected account (child company) owned by a parent
     * company.
     */
    createAPIKey(parentCompanyID, body, options) {
        return this._client.post((0, path_1.path) `/companies/${parentCompanyID}/api_keys`, { body, ...options });
    }
}
exports.Companies = Companies;
