// @whop/sdk/resources/course-lessons.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CourseLessons = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Course lessons
 */
class CourseLessons extends resource_1.APIResource {
    /**
     * Returns a paginated list of lessons within a course or chapter, ordered by
     * position.
     *
     * Required permissions:
     *
     * - `courses:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const courseLessonListResponse of client.courseLessons.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/course_lessons', (pagination_1.CursorPage), {
            query,
            ...options,
        });
    }
    /**
     * Create a new lesson within a course chapter. Lessons can contain video, text, or
     * assessment content.
     *
     * Required permissions:
     *
     * - `courses:update`
     *
     * @example
     * ```ts
     * const lesson = await client.courseLessons.create({
     *   chapter_id: 'chap_xxxxxxxxxxxxx',
     *   lesson_type: 'text',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/course_lessons', { body, ...options });
    }
    /**
     * Retrieves the details of an existing course lesson.
     *
     * Required permissions:
     *
     * - `courses:read`
     *
     * @example
     * ```ts
     * const lesson = await client.courseLessons.retrieve(
     *   'lesn_xxxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/course_lessons/${id}`, options);
    }
    /**
     * Update a lesson's content, type, visibility, assessment questions, or media
     * attachments.
     *
     * Required permissions:
     *
     * - `courses:update`
     *
     * @example
     * ```ts
     * const lesson = await client.courseLessons.update(
     *   'lesn_xxxxxxxxxxxxx',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/course_lessons/${id}`, { body, ...options });
    }
    /**
     * Permanently delete a lesson and remove it from its chapter.
     *
     * Required permissions:
     *
     * - `courses:update`
     *
     * @example
     * ```ts
     * const courseLesson = await client.courseLessons.delete(
     *   'lesn_xxxxxxxxxxxxx',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/course_lessons/${id}`, options);
    }
    /**
     * Mark a lesson as completed for the current user after they finish the content.
     *
     * @example
     * ```ts
     * const response = await client.courseLessons.markAsCompleted(
     *   'lesson_id',
     * );
     * ```
     */
    markAsCompleted(lessonID, options) {
        return this._client.post((0, path_1.path) `/course_lessons/${lessonID}/mark_as_completed`, options);
    }
    /**
     * Record that the current user has started viewing a lesson, creating progress
     * tracking records.
     *
     * @example
     * ```ts
     * const response = await client.courseLessons.start(
     *   'lesson_id',
     * );
     * ```
     */
    start(lessonID, options) {
        return this._client.post((0, path_1.path) `/course_lessons/${lessonID}/start`, options);
    }
    /**
     * Submit answers for a quiz or knowledge check lesson and receive a graded result.
     *
     * @example
     * ```ts
     * const response =
     *   await client.courseLessons.submitAssessment('lesson_id', {
     *     answers: [{ question_id: 'question_id' }],
     *   });
     * ```
     */
    submitAssessment(lessonID, body, options) {
        return this._client.post((0, path_1.path) `/course_lessons/${lessonID}/submit_assessment`, { body, ...options });
    }
}
exports.CourseLessons = CourseLessons;
