// @whop/sdk/resources/dm-channels.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Dm channels
 */
export class DmChannels extends APIResource {
    /**
     * Returns a paginated list of DM channels for the currently authenticated user,
     * sorted by most recently active.
     *
     * Required permissions:
     *
     * - `dms:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const dmChannelListResponse of client.dmChannels.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/dm_channels', (CursorPage), { query, ...options });
    }
    /**
     * Create a new DM channel between two or more users, optionally scoped to a
     * specific company. Returns the existing channel if one already exists.
     *
     * @example
     * ```ts
     * const dmChannel = await client.dmChannels.create({
     *   with_user_ids: ['string'],
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/dm_channels', { body, ...options });
    }
    /**
     * Retrieves the details of an existing DM channel.
     *
     * Required permissions:
     *
     * - `dms:read`
     *
     * @example
     * ```ts
     * const dmChannel = await client.dmChannels.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/dm_channels/${id}`, options);
    }
    /**
     * Update the settings of an existing DM channel, such as its display name. Only an
     * admin of the channel can perform this action.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmChannel = await client.dmChannels.update('id');
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/dm_channels/${id}`, { body, ...options });
    }
    /**
     * Permanently delete a DM channel and all of its messages. Only an admin of the
     * channel can perform this action.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmChannel = await client.dmChannels.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/dm_channels/${id}`, options);
    }
}
