// @whop/sdk/resources/dm-members.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Dm members
 */
export class DmMembers extends APIResource {
    /**
     * Returns a paginated list of members in a specific DM channel, sorted by the date
     * they were added.
     *
     * Required permissions:
     *
     * - `dms:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const dmMemberListResponse of client.dmMembers.list(
     *   { channel_id: 'channel_id' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(query, options) {
        return this._client.getAPIList('/dm_members', (CursorPage), { query, ...options });
    }
    /**
     * Add a new user to an existing DM channel. Only an admin of the channel can add
     * members.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.create({
     *   channel_id: 'channel_id',
     *   user_id: 'user_xxxxxxxxxxxxx',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/dm_members', { body, ...options });
    }
    /**
     * Retrieves the details of an existing DM member.
     *
     * Required permissions:
     *
     * - `dms:read`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/dm_members/${id}`, options);
    }
    /**
     * Update a DM channel member's settings, such as their notification preferences or
     * membership status.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.update('id');
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/dm_members/${id}`, { body, ...options });
    }
    /**
     * Remove a user from a DM channel. An admin can remove any member, and a member
     * can remove themselves.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/dm_members/${id}`, options);
    }
}
