// @whop/sdk/resources/dm-members.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DmMembers = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Dm members
 */
class DmMembers extends resource_1.APIResource {
    /**
     * Returns a paginated list of members in a specific DM channel, sorted by the date
     * they were added.
     *
     * Required permissions:
     *
     * - `dms:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const dmMemberListResponse of client.dmMembers.list(
     *   { channel_id: 'channel_id' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(query, options) {
        return this._client.getAPIList('/dm_members', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Add a new user to an existing DM channel. Only an admin of the channel can add
     * members.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.create({
     *   channel_id: 'channel_id',
     *   user_id: 'user_xxxxxxxxxxxxx',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/dm_members', { body, ...options });
    }
    /**
     * Retrieves the details of an existing DM member.
     *
     * Required permissions:
     *
     * - `dms:read`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/dm_members/${id}`, options);
    }
    /**
     * Update a DM channel member's settings, such as their notification preferences or
     * membership status.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.update('id');
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/dm_members/${id}`, { body, ...options });
    }
    /**
     * Remove a user from a DM channel. An admin can remove any member, and a member
     * can remove themselves.
     *
     * Required permissions:
     *
     * - `dms:channel:manage`
     *
     * @example
     * ```ts
     * const dmMember = await client.dmMembers.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/dm_members/${id}`, options);
    }
}
exports.DmMembers = DmMembers;
