// @whop/sdk/resources/invoices.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Invoices = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Invoices
 */
class Invoices extends resource_1.APIResource {
    /**
     * Returns a paginated list of invoices for a company, with optional filtering by
     * product, status, collection method, and creation date.
     *
     * Required permissions:
     *
     * - `invoice:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const invoiceListItem of client.invoices.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/invoices', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Create an invoice for a customer. The invoice can be charged automatically using
     * a stored payment method, or sent to the customer for manual payment.
     *
     * Required permissions:
     *
     * - `invoice:create`
     *
     * @example
     * ```ts
     * const invoice = await client.invoices.create({
     *   collection_method: 'send_invoice',
     *   company_id: 'biz_xxxxxxxxxxxxxx',
     *   plan: {},
     *   product: { title: 'title' },
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/invoices', { body, ...options });
    }
    /**
     * Retrieves the details of an existing invoice.
     *
     * Required permissions:
     *
     * - `invoice:basic:read`
     *
     * @example
     * ```ts
     * const invoice = await client.invoices.retrieve(
     *   'inv_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/invoices/${id}`, options);
    }
    /**
     * Void an open invoice so it can no longer be paid. Voiding is permanent and
     * cannot be undone.
     *
     * Required permissions:
     *
     * - `invoice:update`
     *
     * @example
     * ```ts
     * const response = await client.invoices.void(
     *   'inv_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    void(id, options) {
        return this._client.post((0, path_1.path) `/invoices/${id}/void`, options);
    }
    /**
     * Mark an open invoice as paid when payment was collected outside of Whop.
     *
     * Required permissions:
     *
     * - `invoice:update`
     *
     * @example
     * ```ts
     * const response = await client.invoices.markPaid(
     *   'inv_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    markPaid(id, options) {
        return this._client.post((0, path_1.path) `/invoices/${id}/mark_paid`, options);
    }
    /**
     * Mark an open invoice as uncollectible when payment is not expected.
     *
     * Required permissions:
     *
     * - `invoice:update`
     *
     * @example
     * ```ts
     * const response = await client.invoices.markUncollectible(
     *   'inv_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    markUncollectible(id, options) {
        return this._client.post((0, path_1.path) `/invoices/${id}/mark_uncollectible`, options);
    }
    /**
     * Update a draft invoice's details.
     *
     * Required permissions:
     *
     * - `invoice:update`
     *
     * @example
     * ```ts
     * const invoice = await client.invoices.update(
     *   'inv_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/invoices/${id}`, { body, ...options });
    }
    /**
     * Delete a draft invoice.
     *
     * Required permissions:
     *
     * - `invoice:update`
     *
     * @example
     * ```ts
     * const invoice = await client.invoices.delete(
     *   'inv_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/invoices/${id}`, options);
    }
}
exports.Invoices = Invoices;
