// @whop/sdk/resources/memberships.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Memberships
 */
export class Memberships extends APIResource {
    /**
     * Returns a paginated list of memberships, with optional filtering by product,
     * plan, status, and user.
     *
     * Required permissions:
     *
     * - `member:basic:read`
     * - `member:email:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const membershipListResponse of client.memberships.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/memberships', (CursorPage), { query, ...options });
    }
    /**
     * Retrieves the details of an existing membership.
     *
     * Required permissions:
     *
     * - `member:basic:read`
     * - `member:email:read`
     *
     * @example
     * ```ts
     * const membership = await client.memberships.retrieve(
     *   'mem_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/memberships/${id}`, options);
    }
    /**
     * Update a membership's metadata or other mutable properties.
     *
     * Required permissions:
     *
     * - `member:manage`
     * - `member:email:read`
     * - `member:basic:read`
     *
     * @example
     * ```ts
     * const membership = await client.memberships.update(
     *   'mem_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/memberships/${id}`, { body, ...options });
    }
    /**
     * Cancel a membership either immediately or at the end of the current billing
     * period. Immediate cancellation revokes access right away.
     *
     * Required permissions:
     *
     * - `membership:cancel`
     * - `member:email:read`
     * - `member:basic:read`
     *
     * @example
     * ```ts
     * const membership = await client.memberships.cancel(
     *   'mem_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    cancel(id, body = {}, options) {
        return this._client.post(path `/memberships/${id}/cancel`, { body, ...options });
    }
    /**
     * Pause a membership's recurring payments. The customer retains access but will
     * not be charged until the membership is resumed.
     *
     * Required permissions:
     *
     * - `member:manage`
     * - `member:email:read`
     * - `member:basic:read`
     *
     * @example
     * ```ts
     * const membership = await client.memberships.pause(
     *   'mem_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    pause(id, body = {}, options) {
        return this._client.post(path `/memberships/${id}/pause`, { body, ...options });
    }
    /**
     * Resume a previously paused membership's recurring payments. Billing resumes on
     * the next cycle.
     *
     * Required permissions:
     *
     * - `member:manage`
     * - `member:email:read`
     * - `member:basic:read`
     *
     * @example
     * ```ts
     * const membership = await client.memberships.resume(
     *   'mem_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    resume(id, options) {
        return this._client.post(path `/memberships/${id}/resume`, options);
    }
    /**
     * Reverse a pending cancellation for a membership that was scheduled to cancel at
     * period end.
     *
     * Required permissions:
     *
     * - `member:manage`
     * - `member:email:read`
     * - `member:basic:read`
     *
     * @example
     * ```ts
     * const membership = await client.memberships.uncancel(
     *   'mem_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    uncancel(id, options) {
        return this._client.post(path `/memberships/${id}/uncancel`, options);
    }
    /**
     * Add free days to extend a membership's current billing period, expiration date,
     * or Stripe trial.
     *
     * Required permissions:
     *
     * - `member:manage`
     * - `member:email:read`
     * - `member:basic:read`
     *
     * @example
     * ```ts
     * const membership = await client.memberships.addFreeDays(
     *   'mem_xxxxxxxxxxxxxx',
     *   { free_days: 42 },
     * );
     * ```
     */
    addFreeDays(id, body, options) {
        return this._client.post(path `/memberships/${id}/add_free_days`, { body, ...options });
    }
}
