// @whop/sdk/resources/messages.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Messages = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Messages
 */
class Messages extends resource_1.APIResource {
    /**
     * Returns a paginated list of messages within a specific experience chat, DM, or
     * group chat channel, sorted by creation time.
     *
     * Required permissions:
     *
     * - `chat:read`
     */
    list(query, options) {
        return this._client.getAPIList('/messages', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Retrieves the details of an existing message.
     *
     * Required permissions:
     *
     * - `chat:read`
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/messages/${id}`, options);
    }
    /**
     * Send a new message in an experience chat, DM, or group chat channel. Supports
     * text content, attachments, polls, and replies.
     *
     * Required permissions:
     *
     * - `chat:message:create`
     */
    create(body, options) {
        return this._client.post('/messages', { body, ...options });
    }
    /**
     * Edit the content, attachments, or pinned status of an existing message in an
     * experience chat, DM, or group chat channel.
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/messages/${id}`, { body, ...options });
    }
    /**
     * Permanently delete a message from an experience chat, DM, or group chat channel.
     * Only the message author or a channel admin can delete a message.
     *
     * Required permissions:
     *
     * - `chat:message:create`
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/messages/${id}`, options);
    }
}
exports.Messages = Messages;
