// @whop/sdk/resources/messages.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Messages
 */
export class Messages extends APIResource {
    /**
     * Returns a paginated list of messages within a specific experience chat, DM, or
     * group chat channel, sorted by creation time.
     *
     * Required permissions:
     *
     * - `chat:read`
     */
    list(query, options) {
        return this._client.getAPIList('/messages', (CursorPage), { query, ...options });
    }
    /**
     * Retrieves the details of an existing message.
     *
     * Required permissions:
     *
     * - `chat:read`
     */
    retrieve(id, options) {
        return this._client.get(path `/messages/${id}`, options);
    }
    /**
     * Send a new message in an experience chat, DM, or group chat channel. Supports
     * text content, attachments, polls, and replies.
     *
     * Required permissions:
     *
     * - `chat:message:create`
     */
    create(body, options) {
        return this._client.post('/messages', { body, ...options });
    }
    /**
     * Edit the content, attachments, or pinned status of an existing message in an
     * experience chat, DM, or group chat channel.
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/messages/${id}`, { body, ...options });
    }
    /**
     * Permanently delete a message from an experience chat, DM, or group chat channel.
     * Only the message author or a channel admin can delete a message.
     *
     * Required permissions:
     *
     * - `chat:message:create`
     */
    delete(id, options) {
        return this._client.delete(path `/messages/${id}`, options);
    }
}
