// @whop/sdk/resources/payments.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { CursorPage } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Payments
 */
export class Payments extends APIResource {
    /**
     * Returns a paginated list of payments for the actor in context, with optional
     * filtering by product, plan, status, billing reason, currency, and creation date.
     *
     * Required permissions:
     *
     * - `payment:basic:read`
     * - `plan:basic:read`
     * - `access_pass:basic:read`
     * - `member:email:read`
     * - `member:basic:read`
     * - `member:phone:read`
     * - `promo_code:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const paymentListResponse of client.payments.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/payments', (CursorPage), { query, ...options });
    }
    /**
     * Retrieves the details of an existing payment.
     *
     * Required permissions:
     *
     * - `payment:basic:read`
     * - `plan:basic:read`
     * - `access_pass:basic:read`
     * - `member:email:read`
     * - `member:basic:read`
     * - `member:phone:read`
     * - `promo_code:basic:read`
     * - `payment:dispute:read`
     * - `payment:resolution_center_case:read`
     *
     * @example
     * ```ts
     * const payment = await client.payments.retrieve(
     *   'pay_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/payments/${id}`, options);
    }
    /**
     * Issue a full or partial refund for a payment. The refund is processed through
     * the original payment processor and the membership status is updated accordingly.
     *
     * Required permissions:
     *
     * - `payment:manage`
     * - `plan:basic:read`
     * - `access_pass:basic:read`
     * - `member:email:read`
     * - `member:basic:read`
     * - `member:phone:read`
     * - `promo_code:basic:read`
     * - `payment:dispute:read`
     * - `payment:resolution_center_case:read`
     *
     * @example
     * ```ts
     * const payment = await client.payments.refund(
     *   'pay_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    refund(id, body = {}, options) {
        return this._client.post(path `/payments/${id}/refund`, { body, ...options });
    }
    /**
     * Retry a failed or pending payment. This re-attempts the charge using the
     * original payment method and plan details.
     *
     * Required permissions:
     *
     * - `payment:manage`
     * - `plan:basic:read`
     * - `access_pass:basic:read`
     * - `member:email:read`
     * - `member:basic:read`
     * - `member:phone:read`
     * - `promo_code:basic:read`
     * - `payment:dispute:read`
     * - `payment:resolution_center_case:read`
     *
     * @example
     * ```ts
     * const payment = await client.payments.retry(
     *   'pay_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    retry(id, options) {
        return this._client.post(path `/payments/${id}/retry`, options);
    }
    /**
     * Void a payment that has not yet been settled. Voiding cancels the payment before
     * it is captured by the payment processor.
     *
     * Required permissions:
     *
     * - `payment:manage`
     * - `plan:basic:read`
     * - `access_pass:basic:read`
     * - `member:email:read`
     * - `member:basic:read`
     * - `member:phone:read`
     * - `promo_code:basic:read`
     * - `payment:dispute:read`
     * - `payment:resolution_center_case:read`
     *
     * @example
     * ```ts
     * const payment = await client.payments.void(
     *   'pay_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    void(id, options) {
        return this._client.post(path `/payments/${id}/void`, options);
    }
    /**
     * Charge an existing member off-session using one of their stored payment methods.
     * You can provide an existing plan, or create a new one in-line. This endpoint
     * will respond with a payment object immediately, but the payment is processed
     * asynchronously in the background. Use webhooks to be notified when the payment
     * succeeds or fails.
     *
     * Required permissions:
     *
     * - `payment:charge`
     * - `plan:create`
     * - `access_pass:create`
     * - `access_pass:update`
     * - `plan:basic:read`
     * - `access_pass:basic:read`
     * - `member:email:read`
     * - `member:basic:read`
     * - `member:phone:read`
     * - `promo_code:basic:read`
     * - `payment:dispute:read`
     * - `payment:resolution_center_case:read`
     *
     * @example
     * ```ts
     * const payment = await client.payments.create({
     *   company_id: 'biz_xxxxxxxxxxxxxx',
     *   member_id: 'mber_xxxxxxxxxxxxx',
     *   payment_method_id: 'pmt_xxxxxxxxxxxxxx',
     *   plan: { currency: 'usd' },
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/payments', { body, ...options });
    }
    /**
     * Returns the list of fees associated with a specific payment, including platform
     * fees and processing fees.
     *
     * Required permissions:
     *
     * - `payment:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const paymentListFeesResponse of client.payments.listFees(
     *   'pay_xxxxxxxxxxxxxx',
     * )) {
     *   // ...
     * }
     * ```
     */
    listFees(id, query = {}, options) {
        return this._client.getAPIList(path `/payments/${id}/fees`, (CursorPage), {
            query,
            ...options,
        });
    }
}
