// @whop/sdk/resources/affiliates/affiliates.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Affiliates = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const OverridesAPI = tslib_1.__importStar(require("./overrides.js"));
const overrides_1 = require("./overrides.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Affiliates
 */
class Affiliates extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.overrides = new OverridesAPI.Overrides(this._client);
    }
    /**
     * Returns a paginated list of affiliates for the actor in context, with optional
     * filtering by status, search, and sorting.
     *
     * Required permissions:
     *
     * - `affiliate:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const affiliateListResponse of client.affiliates.list(
     *   { company_id: 'biz_xxxxxxxxxxxxxx' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(query, options) {
        return this._client.getAPIList('/affiliates', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Creates or finds an affiliate for a company and user.
     *
     * Required permissions:
     *
     * - `affiliate:create`
     *
     * @example
     * ```ts
     * const affiliate = await client.affiliates.create({
     *   company_id: 'biz_xxxxxxxxxxxxxx',
     *   user_identifier: 'user_identifier',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/affiliates', { body, ...options });
    }
    /**
     * Retrieves the details of an existing affiliate.
     *
     * Required permissions:
     *
     * - `affiliate:basic:read`
     *
     * @example
     * ```ts
     * const affiliate = await client.affiliates.retrieve(
     *   'aff_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/affiliates/${id}`, options);
    }
    /**
     * Archives an existing Affiliate
     *
     * Required permissions:
     *
     * - `affiliate:update`
     *
     * @example
     * ```ts
     * const response = await client.affiliates.archive(
     *   'aff_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    archive(id, options) {
        return this._client.post((0, path_1.path) `/affiliates/${id}/archive`, options);
    }
    /**
     * Unarchives an existing Affiliate
     *
     * Required permissions:
     *
     * - `affiliate:update`
     *
     * @example
     * ```ts
     * const response = await client.affiliates.unarchive(
     *   'aff_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    unarchive(id, options) {
        return this._client.post((0, path_1.path) `/affiliates/${id}/unarchive`, options);
    }
}
exports.Affiliates = Affiliates;
Affiliates.Overrides = overrides_1.Overrides;
