// @whop/sdk/resources/affiliates/affiliates.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as OverridesAPI from "./overrides.mjs";
import { Overrides, } from "./overrides.mjs";
import { CursorPage } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Affiliates
 */
export class Affiliates extends APIResource {
    constructor() {
        super(...arguments);
        this.overrides = new OverridesAPI.Overrides(this._client);
    }
    /**
     * Returns a paginated list of affiliates for the actor in context, with optional
     * filtering by status, search, and sorting.
     *
     * Required permissions:
     *
     * - `affiliate:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const affiliateListResponse of client.affiliates.list(
     *   { company_id: 'biz_xxxxxxxxxxxxxx' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(query, options) {
        return this._client.getAPIList('/affiliates', (CursorPage), { query, ...options });
    }
    /**
     * Creates or finds an affiliate for a company and user.
     *
     * Required permissions:
     *
     * - `affiliate:create`
     *
     * @example
     * ```ts
     * const affiliate = await client.affiliates.create({
     *   company_id: 'biz_xxxxxxxxxxxxxx',
     *   user_identifier: 'user_identifier',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/affiliates', { body, ...options });
    }
    /**
     * Retrieves the details of an existing affiliate.
     *
     * Required permissions:
     *
     * - `affiliate:basic:read`
     *
     * @example
     * ```ts
     * const affiliate = await client.affiliates.retrieve(
     *   'aff_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/affiliates/${id}`, options);
    }
    /**
     * Archives an existing Affiliate
     *
     * Required permissions:
     *
     * - `affiliate:update`
     *
     * @example
     * ```ts
     * const response = await client.affiliates.archive(
     *   'aff_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    archive(id, options) {
        return this._client.post(path `/affiliates/${id}/archive`, options);
    }
    /**
     * Unarchives an existing Affiliate
     *
     * Required permissions:
     *
     * - `affiliate:update`
     *
     * @example
     * ```ts
     * const response = await client.affiliates.unarchive(
     *   'aff_xxxxxxxxxxxxxx',
     * );
     * ```
     */
    unarchive(id, options) {
        return this._client.post(path `/affiliates/${id}/unarchive`, options);
    }
}
Affiliates.Overrides = Overrides;
