// @whop/sdk/resources/affiliates/overrides.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { CursorPage } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Affiliates
 */
export class Overrides extends APIResource {
    /**
     * Returns a paginated list of overrides for an affiliate.
     *
     * Required permissions:
     *
     * - `affiliate:basic:read`
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const overrideListResponse of client.affiliates.overrides.list(
     *   'aff_xxxxxxxxxxxxxx',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(id, query = {}, options) {
        return this._client.getAPIList(path `/affiliates/${id}/overrides`, (CursorPage), {
            query,
            ...options,
        });
    }
    /**
     * Creates a commission override for an affiliate.
     *
     * Required permissions:
     *
     * - `affiliate:create`
     *
     * @example
     * ```ts
     * const override = await client.affiliates.overrides.create(
     *   'aff_xxxxxxxxxxxxxx',
     *   {
     *     id: 'id',
     *     commission_value: 6.9,
     *     override_type: 'standard',
     *     plan_id: 'plan_xxxxxxxxxxxxx',
     *   },
     * );
     * ```
     */
    create(id, body, options) {
        return this._client.post(path `/affiliates/${id}/overrides`, { body, ...options });
    }
    /**
     * Retrieves the details of a specific affiliate override.
     *
     * Required permissions:
     *
     * - `affiliate:basic:read`
     *
     * @example
     * ```ts
     * const override = await client.affiliates.overrides.retrieve(
     *   'override_id',
     *   { id: 'aff_xxxxxxxxxxxxxx' },
     * );
     * ```
     */
    retrieve(overrideID, params, options) {
        const { id } = params;
        return this._client.get(path `/affiliates/${id}/overrides/${overrideID}`, options);
    }
    /**
     * Updates an existing affiliate override.
     *
     * Required permissions:
     *
     * - `affiliate:update`
     *
     * @example
     * ```ts
     * const override = await client.affiliates.overrides.update(
     *   'override_id',
     *   { id: 'aff_xxxxxxxxxxxxxx' },
     * );
     * ```
     */
    update(overrideID, params, options) {
        const { id, ...body } = params;
        return this._client.patch(path `/affiliates/${id}/overrides/${overrideID}`, { body, ...options });
    }
    /**
     * Deletes an affiliate override.
     *
     * Required permissions:
     *
     * - `affiliate:update`
     *
     * @example
     * ```ts
     * const override = await client.affiliates.overrides.delete(
     *   'override_id',
     *   { id: 'aff_xxxxxxxxxxxxxx' },
     * );
     * ```
     */
    delete(overrideID, params, options) {
        const { id } = params;
        return this._client.delete(path `/affiliates/${id}/overrides/${overrideID}`, options);
    }
}
